### PR TITLE
remove using fixed IP from tests with SSL

### PR DESCRIPTION
**Pull request by CheSema**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
No need to use fixed IP when you are using cluster with SSL. Using the same fixed IP could trigger collision between  tests.
At this change the server's certificate is generated for a designated host name (see server-ext.cnf at each test). The client should check server's certificate against that name accordingly.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/integration/parallel_skip.json [hunk 1]
```diff
@@ -48,17 +48,6 @@
   "test_system_replicated_fetches/test.py::test_system_replicated_fetches",
   "test_zookeeper_config_load_balancing/test.py::test_round_robin",
 
-  "test_tlsv1_3/test.py::test_https",
-  "test_tlsv1_3/test.py::test_https_wrong_cert",
-  "test_tlsv1_3/test.py::test_https_non_ssl_auth",
-  "test_tlsv1_3/test.py::test_create_user",
   "test_user_ip_restrictions/test.py::test_ipv4",
   "test_user_ip_restrictions/test.py::test_ipv6",
```
Comment (tavplubix) [inline tests/integration/parallel_skip.json:52]: AFAIK, json does not allow trailing comma

File: tests/integration/test_tlsv1_3/test.py [hunk 2]
```diff
@@ -1,19 +1,20 @@
 import pytest
 from helpers.cluster import ClickHouseCluster
+from helpers.ssl_context import WrapSSLContextWithSNI
 import urllib.request, urllib.parse
 import ssl
 import os.path
 
+
+# The test cluster is configured with certificate for that host name, see 'server-cert.pem'.
+# The client have to verify server certificate against that name. Client uses SNI
```
Comment (tavplubix) [inline tests/integration/test_tlsv1_3/test.py:10]:
```suggestion
# The client has to verify server certificate against that name. Client uses SNI
```

**Review by tavplubix — APPROVED** [review]
LGTM, but should we leave one test without SNI just in case?

File: tests/integration/parallel_skip.json (hunk 1 above)
Comment (CheSema) [reply tests/integration/parallel_skip.json]: You are right. Fixed.

**Comment by CheSema** [comment]
> LGTM, but should we leave one test without SNI just in case?

Don't see the cases which require that.
 
We do set ssl_host explicitly in the same way when client connects to the server. That option has effect only at SSL handshake. Client check the certificate against some host name. Handshake just fails if the host is unexpected.
https://clickhouse.com/codebrowser/ClickHouse/src/Client/Connection.cpp.html#113
It should be treated like default behaviour. You connect to the server, you know the server (whatever how), you check it's certificate against your knowledge. 

For me, more important is to preserve all the checks with option like this: context.check_hostname = True. I have seen PR where that option is relaxed. So I think we should

**Pull request merged**